src/frontend/src/components/modal.ts

```diff
@@ -7,78 +7,47 @@ import { withRef } from "../utils/lit-html";
  **/
 export const createModal = ({
   slot,
-  closeButtonRef,
-  submitButtonRef,
+  submit,
 }: {
   slot: TemplateResult;
-  closeButtonRef: Ref<HTMLButtonElement>;
-  submitButtonRef: Ref<HTMLButtonElement>;
-}): HTMLDialogElement => {
+  submit: () => void;
+}) => {
+  const modalElement: Ref<HTMLDialogElement> = createRef();
+
+  const closeModal = () =>
+    withRef(modalElement, (modalElement) => {
+      modalElement.close();
+      modalElement.remove();
+    });
+
   const modalHtml = html`
-    <form method="dialog" class="c-modal__content c-card c-card--modal">
-      <div class="c-modal__inner">
-        <button
-          class="c-modal__close"
-          aria-label="Close Modal"
-          ${ref(closeButtonRef)}
-        >
-          &times;
-        </button>
-        ${slot}
-        <div class="c-modal__footer">
+    <dialog class="c-modal" aria-modal ${ref(modalElement)}>
```

Comment:
@peterpeterparker `aria-modal` is equivalent to `aria-modal="true"`, right?

Reply:
Good question, not sure. The default value is `false`.

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-modal

Reply:
> Boolean attributes are considered to be true if they're present on the element at all.
> https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute

all good

Reply:
Great!